### PR TITLE
Updates/ad reporting v1.0.0

### PR DESCRIPTION
**Pull request by fivetran-sheringuyen**
**Are you a current Fivetran customer?** 

Fivetran Employee

**What change(s) does this PR introduce?** 

## 🚨 Breaking Changes 🚨
- The previous `ad_reporting` model has been renamed to `ad_reporting__url_report` and will only include records that have non-null url values for more information on specific filters please refer to each platform package's `url_report` model.

## 🎉 Feature Enhancements 🎉 
- Apple Search Ads has officially been released and added to Ad Reporting.
- In addition to the `ad_reporting__url_report` model update, we have added **five** new models:
  - `ad_reporting__account_report`
  - `ad_reporting__campaign_report`
  - `ad_reporting__ad_group_report`
  - `ad_reporting__ad_report`
  - `ad_reporting__keyword_search_report`

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [x] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

The previous `ad_reporting` model has been renamed to `ad_reporting__url_report` and will only include records that have non-null url values for more information on specific filters please refer to each platform package's `url_report` model.

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [x] No 

**How did you test the PR changes?** 

 

- [x] CircleCi  
- [x] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [x] BigQuery
- [x] Redshift
- [x] Snowflake
- [x] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
:dancer:

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Comment by fivetran-sheringuyen** [comment]
@fivetran-joemarkiewicz thanks in advance for the PR review. I wanted to leave some specifics I'd like you to take a look at when you start your review and weigh in on:

- [ ] including/excluding search report 
- [ ] README sections specifically
 - [ ] in `STEP 1 PRE-REQS` for calling out FB ads
 - [ ] in `STEP 4` Enabling/disabling models
- [ ] Should there be more documentation for `macros/field_name_conversion.sql`

File: dbt_project.yml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 name: 'ad_reporting'
-version: '0.8.0'
+version: '1.0.0'
 config-version: 2
 
 require-dbt-version: [">=1.0.0", "<2.0.0"]
 
 vars:
+  ad_reporting__apple_search_ads_enabled: true
```
Comment (fivetran-joemarkiewicz) [inline dbt_project.yml:8]: @fivetran-jamie actually uncovered that we should remove these and rely on the inline default values as they cause conflicts in the source. Can you remove these variables.

File: dbt_project.yml [hunk 2]
```diff
@@ -14,41 +15,38 @@ vars:
...
+
 models:
   ad_reporting:
     +schema: ad_reporting
     +materialized: view
-  
-  pinterest:
+
+# remove once `config` settings are incorporated into upstream platform models  
+  facebook_ads:
```
Comment (fivetran-joemarkiewicz) [inline dbt_project.yml:27]: With the config settings applied I think we are good to remove these.

File: models/intermediate/int_ad_reporting__account_report.sql [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                'account_name', 
+                'clicks', 
+                'impressions', 
+                'spend']) }}
+), 
+
+prep_standardized_union_platform_rename as (
+
+    select 
+        cast(date_day as DATE) as date_day,
```
Comment (fivetran-joemarkiewicz) [inline models/intermediate/int_ad_reporting__account_report.sql:22]: I am not sure if `DATE` is universal across platforms. Is there a util we should be using instead?

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
I have a few comments to start below. I also noticed a number of compilation errors within the search term report. Finally, it looks like we don't have all of the platforms included in the end reports. Are these up to date, or are the comments left overs from before?

File: README.md [hunk 4]
```diff
@@ -214,39 +180,115 @@ models:
...
+</details>
+<br>
 
-### Databricks Dispatch Configuration
-dbt `v0.20.0` introduced a new project-level dispatch configuration that enables an "override" setting for all dispatched macros. If you are using a Databricks destination with this package you will need to add the below (or a variation of the below) dispatch configuration within your `dbt_project.yml`. This is required in order for the package to accurately search for macros within the `dbt-labs/spark_utils` then the `dbt-labs/dbt_utils` packages respectively.
+# 🔍 Does this package have dependencies?
+This dbt package is dependent on the following dbt packages. For more information on the below packages, refer to the [dbt hub](https://hub.getdbt.com/) site.
+> **If you have any of these dependent packages in your own `packages.yml` I highly recommend you remove them to ensure there are no package version conflicts.**
+
+# !!! TO DO: DOUBLE CHECK VERSIONS ON RELEASE!!!!
```
Comment (fivetran-joemarkiewicz) [inline README.md:216]: We should be able to update these now!

Comment (fivetran-joemarkiewicz) [reply README.md:216]: Also, [dbt-expecations](https://hub.getdbt.com/calogica/dbt_expectations/latest/) is used in google ads and should probably be included here as well.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-sheringuyen this is a herculean update and you have absolutely crushed it!!! 💪 I tried to stress test the macros and was incredibly impressed with the work put into these. I feel a few of them can ultimately move to Fivetran Utils, but that can be a task for a future release. My only major comment with the macros is to add documentation for them within the package for community awareness and our own benefit for future enhancements. 

I have a few small comments and suggestions inline and below that we should apply before we merge and release. Once those are applied, we will be good to go for our first v1.0.0 release!
- [x] Can you add the `.DS_Store` files to the respective `.gitignore` files. They are not needed and can be ignored in the final release.
- [x] Can you add documentation for the macros you have added. After your walkthrough of the macros (and testing them) I have a good understanding of what they are doing. However, it would be great if we could create a `.yml` file in the macros folder to document the macros and the respective arguments. You can follow the guide shown [in the dbt docs](https://docs.getdbt.com/faqs/docs/documenting-macros) on how to document macros.
- [x] Reminder to update the packages.yml
- [x] Address any other comments in-line of this review
- [x] Update the docs one last time before release.

File: dbt_project.yml [hunk 5]
```diff
@@ -1,62 +1,11 @@
...
-  ad_reporting__google_ads_enabled: true
-  ad_reporting__twitter_ads_enabled: true
-  ad_reporting__facebook_ads_enabled: true
-  ad_reporting__snapchat_ads_enabled: true
-  ad_reporting__tiktok_ads_enabled: true
 
 models:
   ad_reporting:
     +schema: ad_reporting
     +materialized: view
```
[the remark this answers is not in the document]
Comment (fivetran-sheringuyen) [reply dbt_project.yml:11]: This was something I noticed as well and forgot to mention during our talks. Thanks for bringing this up! I was actually considering having the final models be tables instead and the intermediate models being either views or ephemeral. I personally don't have a preference and would love to get the team's input on this!

File: README.md (hunk 4 above)
Comment (fivetran-sheringuyen) [reply README.md:216]: Thanks for catching - I also filed an issue for `google_ads_source` to add that as a dependency there as well!

File: dbt_project.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (fivetran-sheringuyen) [reply dbt_project.yml:11]: Update - I think I am going to go toward the path of: final models -> tables & intermediate models  -> ephemeral given that the intermediate models are actually quite a bit in terms of logic (not super complex in logic but may be complex in quantity).

[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply dbt_project.yml:11]: That plan sounds good to me!

**Pull request merged**